src/Type/WebMozartAssert/AssertTypeSpecifyingExtension.php

```diff
@@ -667,6 +656,44 @@ static function (Type $type) use ($valueType): Type {
 		throw new ShouldNotHappenException();
 	}
 
+	private function handleAll(
+		string $methodName,
+		StaticCall $node,
+		Scope $scope
+	): SpecifiedTypes
+	{
+		if (!$scope instanceof MutatingScope) {
+			throw new ShouldNotHappenException();
+		}
+
+		$typeBefore = $scope->getType($node->getArgs()[0]->value);
+		if (!$typeBefore->isIterable()->yes()) {
+			return new SpecifiedTypes([], []);
+		}
+
+		$variable = new Variable('item');
+		// @phpstan-ignore-next-line assignVariable is not covered by BC promise
+		$scope = $scope->assignVariable('item', $typeBefore->getIterableValueType());
+
+		$args = $node->getArgs();
+		$args[0] = new Arg($variable);
```

Comment:
An idea - what about `new Arg(new GetIterableValueTypeExpr($node->getArgs()[0]->value))`? It might crash somewhere as these virtual expr nodes are experimental, but it avoids the whole MutatingScope business.

Reply:
I re-used parts of my first solution combined with the SpecifiedTypes normalization which makes more sense I guess